python/ray/tests/test_autoscaler.py

```diff
@@ -2239,14 +2239,13 @@ def testConfiguresNewNodes(self):
         )
 
         autoscaler.update()
```

Comment:
Note:

* `autoscaler.update()` makes autoscaling decisions based on `non_terminated_nodes` and `pending_launches`.
* If `autoscaler.update()` decides to launch nodes, it will put the list of nodes `to_launch` into a queue.
* The background node launcher (another thread) will async consume the queues to launch the node.
* The background node launcher consumes the queue, and remove it from `pending_launches` but before updating `non_terminated_nodes`.
* The second `autoscaler.update()` is called, so a new node is created unexpectedly.